### PR TITLE
Sync to upstream

### PR DESCRIPTION
**Pull request by fanquake**
Pulls the few changes from upstream (last commit ~ 2 years ago).
Seems reasonable to do before making any other changes. i.e https://github.com/bitcoin-core/crc32c-subtree/pull/7, or if we are going to make further changes to the CMake build system etc.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,7 +1,6 @@
 # CRC32C
 
-[![Build Status](https://travis-ci.org/google/crc32c.svg?branch=master)](https://travis-ci.org/google/crc32c)
-[![Build Status](https://ci.appveyor.com/api/projects/status/moiq7331pett4xuj/branch/master?svg=true)](https://ci.appveyor.com/project/pwnall/crc32c)
+[![Build Status](https://github.com/google/crc32c/actions/workflows/build.yml/badge.svg)](https://github.com/google/crc32c/actions/workflows/build.yml)
```
Comment (willcl-ark) [inline README.md:3]: This isn't correct for us, but can be fixed later

**Review by willcl-ark — COMMENTED** [review]
The CI + CMakeLists.txt as they stand here are not worth adding: https://github.com/willcl-ark/crc32c-subtree/actions/runs/10605521278/job/29394471127

But tests and benches can be relatively easily [hacked(1)](https://github.com/bitcoin-core/crc32c-subtree/commit/080d9b5e1facab7c403b25dff4fda813a632546d) [out(2)](https://github.com/bitcoin-core/crc32c-subtree/commit/8e0e1e61eeefb4cc9aac0b71efe7eadd454ee8fe) to get an array of simple build tests passing again: https://github.com/willcl-ark/crc32c-subtree/actions/runs/10605602819

(I assume this is preferable to merging 3rd party deps).

The rest is trivial enough.

**Comment by fanquake** [comment]
The point is just to pull the upstream changes. Any other (relevant) changes could be done in a followup.

**Comment by willcl-ark** [comment]
Yes I agree, in this case it looks fine to me. 👍🏼

Ripping out the third party stuff from the CMakLists and CI is easy enough to do later.

**Pull request merged**